src/util-base64.c

```diff
@@ -366,7 +366,7 @@ static int B64DecodeInCompleteStringWSp(void)
     const char *src = "SGVs bG8 gV29y bGQ";
     const char *fin_str = "Hello Wor";
     TEST_RFC2045(src, fin_str, strlen(fin_str) + 1 /* 12 B in dest_size */, strlen(fin_str),
-            strlen(src) - 3, BASE64_ECODE_OK);
+            strlen(src) - 4, BASE64_ECODE_OK);
```

Comment:
Why this change ?

Reply:
In general, I think it would be nice to use explicit sizes in these tests. These strlen derived sizes make it harder to follow what is happening.

Reply:
Because the initial consumed bytes were incorrect. See below.
Destination size: 10
B64 encoded string: SGVs bG8 gV29y bGQ
Mode: RFC 2045

We process them the following way.
First pass: `SGVs` -> `Hel`; consumed bytes: 4; decoded bytes: 3
Second pass: ` bG8 g` -> `bG8g` (skipped spaces as they're not base64 alphabet and any such chars must be skipped as per RFC 2045) -> `lo `, consumed bytes: 6 + 4; decoded bytes: 3 + 3
Third pass: `V29y` -> `Wor`; consumed bytes: 6 + 4 + 4; decoded bytes: 3 + 3 +3
Fourth pass: ` bGQ` but, there isn't enough data or destination size to decode, so we do not process it.

So, finally, consumed bytes: 6 + 4 + 4 = 14 i.e. strlen(src) - 4 where strlen(src) = 18.